src/main/java/com/salesforce/storm/spout/dynamic/SpoutCoordinator.java

```diff
@@ -158,6 +161,42 @@ public void addVirtualSpout(final DelegateSpout spout) {
     }
 
     /**
+     * Remove a new VirtualSpout from the coordinator. This will signal to the monitor to request that the VirtualSpout
+     * be stopped and ultimately removed.
+     *
+     * This method will blocked until the VirtualSpout has completely stopped.
+     *
+     * @param virtualSpoutIdentifier identifier of the VirtualSpout to be removed.
+     */
+    public void removeVirtualSpout(final VirtualSpoutIdentifier virtualSpoutIdentifier) {
```

Comment:
i wonder if this should be a checked exception?  At the least we should denote it in the javadoc and in the method definition via throws